WordPress/gutenberg
# Heading: Complete TypeScript migration of component

## What?
Completes the migration of `Heading` component to TypeScript.

## Why?
Part of the @wordpress/components's TypeScript migration (https://github.com/WordPress/gutenberg/issues/35744).

## How?
- Moved types to `types.js`
- Refactored test to use TypeScript
- Updated docs

## Testing Instructions
1. Run npm run dev and ensure no typing errors
2. Fire up the Storybook and confirm the `Heading` continues to function as expected
3. Test the various Headings within the block/site editors in Gutenberg still work
